crates/e2e/tests/e2e/cow_amm.rs

```diff
@@ -595,3 +596,685 @@ async fn cow_amm_driver_support(web3: Web3) {
+            Token::FixedBytes(
+                hex::decode("5a28e9363bb942b639270062aa6bb295f434bcdfc42c97267bf003f272060dc9")
+                    .unwrap(),
+            ), // sell_token_source == erc20
+            Token::FixedBytes(
+                hex::decode("5a28e9363bb942b639270062aa6bb295f434bcdfc42c97267bf003f272060dc9")
+                    .unwrap(),
+            ), // buy_token_destination == erc20
+        ]),
+        Token::Tuple(vec![
+            Token::Uint(0.into()), // min_traded_token
+            Token::Address(oracle.address()),
+            Token::Bytes(oracle_data),
+            Token::FixedBytes(APP_DATA.to_vec()),
+        ]),
+    ]);
+
+    // Prepend CoW AMM address to the signature so the settlement contract knows
+    // which contract this signature refers to.
+    let signature = cow_amm
+        .address()
+        .as_bytes()
+        .iter()
+        .cloned()
+        .chain(signature_data)
+        .collect::<Vec<_>>();
+
+    // Create the commitment call for the pre-interaction
+    let cow_amm_commitment = {
+        let order_hash = cow_amm_order.hash_struct();
+        let order_hash = hashed_eip712_message(&onchain.contracts().domain_separator, &order_hash);
+        let commitment = cow_amm
+            .commit(ethcontract::Bytes(order_hash))
+            .tx
+            .data
+            .unwrap();
+        Call {
+            target: cow_amm.address(),
+            value: 0.into(),
+            calldata: commitment.0.to_vec(),
+        }
+    };
+
+    // Fund trader "bob" with DAI and approve allowance
+    dai.mint(bob.address(), to_wei(250)).await;
+    tx!(
+        bob.account(),
+        dai.approve(onchain.contracts().allowance, U256::MAX)
+    );
+
+    // Get balances before the trade
+    let amm_weth_balance_before = onchain
+        .contracts()
+        .weth
+        .balance_of(cow_amm.address())
+        .call()
+        .await
+        .unwrap();
+    let bob_weth_balance_before = onchain
+        .contracts()
+        .weth
+        .balance_of(bob.address())
+        .call()
+        .await
+        .unwrap();
+
+    // Compensate a delay between the `CurrentBlockStream` and the actual onchain
+    // data.
+    tokio::time::sleep(tokio::time::Duration::from_secs(2)).await;
+
+    // Place user order where bob sells DAI to buy WETH (opposite direction)
+    let user_order = OrderCreation {
+        sell_token: dai.address(),
+        sell_amount: to_wei(230), // 230 DAI
+        buy_token: onchain.contracts().weth.address(),
+        buy_amount: U256::exp10(17), // 0.1 WETH
+        valid_to: model::time::now_in_epoch_seconds() + 300,
+        kind: OrderKind::Sell,
+        ..Default::default()
+    }
+    .sign(
+        EcdsaSigningScheme::Eip712,
+        &onchain.contracts().domain_separator,
+        SecretKeyRef::from(&SecretKey::from_slice(bob.private_key()).unwrap()),
+    );
+    let user_order_id = services.create_order(&user_order).await.unwrap();
```

Comment:
To check that the quote actually gets verified it would be good to first get a quote from the API and assert that it has `verified: true`. Then you can call `create_order()` and pass the generated quote id.
Should effectively not change anything but it would make it clear in the test that checking that the quote was verified is important.

Reply:
I forgot to mention that in the description. I wanted to merge these two e2e tests into a single one, but that also requires changes in the mocked solver. The PR is already huge, maybe it makes sense to accomplish this as a follow-up.